### ISSUE TITLE
Add a runtime option to disable self-update

### ISSUE THREAD

**Issue by MatejKafka** [typst/typst#4896]
### Description

Add an environment variable (e.g. `TYPST_NO_SELF_UPDATE`) or a CLI argument (e.g. `--no-self-update`) which disables `typst update` at runtime, similarly to the existing compile-time option.

### Use Case

Many package managers for Windows (WinGet, Chocolatey, Scoop, Pog) use the pre-compiled binaries from GitHub to install Typst. When Typst is managed by the package manager, self-updates will result in inconsistencies, where the package manager believes it installed a different version of Typst than is actually installed. It would be beneficial to allow the package manager to prevent the user from calling `typst update`; the package manager can then expose an executable stub that sets the environment variable or CLI argument and calls the actual Typst binary.

**Comment by Coekjan** [comment]
I think, to solve the problem fundamentally, the package managers should provide binaries **without the `self-update` feature** for their users, instead of requiring typst to provide the proposed env and flag.

**Comment by MatejKafka** [comment]
> I think typst can prebuild two seperate kind of binaries for the release

I agree that would be ideal, if the maintainers are willing to do it.

> those package managers should change their packaging strategy for this app to build it themselves with the correct options for their platform.

None of the well-known Windows package managers do custom builds, for multiple reasons, the most important of which are probably the licensing issues (custom builds must be hosted somewhere, which amounts to redistribution of the original software, and many projects are not happy with redistribution). I understand the expectation, as it's standard for Linux package managers, but it's not done that way on Windows.
